kratos/geometries/triangle_3d_3.h

```diff
@@ -1739,6 +1739,59 @@ template<class TPointType> class Triangle3D3
         return rResult;
     }
 
+    ///@}
+    ///@name Spatial Operations
+    ///@{
+
+    /**
+    * @brief Projects a certain point on the geometry, or finds
+    *        the closest point, depending on the provided
+    *        initial guess. The external point does not necessary
+    *        lay on the geometry.
+    *        It shall deal as the interface to the mathematical
+    *        projection function e.g. the Newton-Raphson.
+    *        Thus, the breaking criteria does not necessarily mean
+    *        that it found a point on the surface, if it is really
+    *        the closest if or not. It shows only if the breaking
+    *        criteria, defined by the tolerance is reached.
+    *
+    *        This function requires an initial guess, provided by
+    *        rProjectedPointLocalCoordinates.
+    *        This function can be a very costly operation.
+    *
+    * @param rPointGlobalCoordinates the point to which the
+    *        projection has to be found.
+    * @param rProjectedPointGlobalCoordinates the location of the
+    *        projection in global coordinates.
+    * @param rProjectedPointLocalCoordinates the location of the
+    *        projection in local coordinates.
+    *        The variable is as initial guess!
+    * @param Tolerance accepted of orthogonal error to projection.
+    * @return It is chosen to take an int as output parameter to
+    *         keep more possibilities within the interface.
+    *         0 -> failed
+    *         1 -> converged
+    */
+    int ProjectionPoint(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+        ) const override
+    {
+
+        PointLocalCoordinates(rProjectedPointLocalCoordinates, rPointGlobalCoordinates);
+
+        for (std::size_t  i = 0; i < 3; i++) {
+            rProjectedPointLocalCoordinates[i] = (rProjectedPointLocalCoordinates[i] < 0.0) ? 0.00 :  rProjectedPointLocalCoordinates[i];
+            rProjectedPointLocalCoordinates[i] = (rProjectedPointLocalCoordinates[i] > 1.0) ? 1.00 :  rProjectedPointLocalCoordinates[i];
+        }
+
+        this->GlobalCoordinates(rProjectedPointGlobalCoordinates, rProjectedPointLocalCoordinates);
+
+        return 1;

```

Comment:
shouldn't this return 0 if it is not inside?

Reply:
* @return It is chosen to take an int as output parameter to
    *         keep more possibilities within the interface.
    *         0 -> failed
    *         1 -> converged

Reply:
That doesn't answer my question...
maybe we need to extend to another parameter but not sure

lets hear the opinion of @tteschemacher 

In any case this fct cannot unconditionally return that it was successful I would say

Reply:
I am just following the API

Reply:
For linear simple cases it actually cannot fail (or at least shouldn't ;) ), accordingly, returning 1 is fine.

If you want to check the burbs geometries, done of them might return 0, here.

One could actually check the input, and if this isn't value it might fail.